src/ImageSharp.Web/FormatUtilities.cs

```diff
@@ -50,21 +52,39 @@ public FormatUtilities(IOptions<ImageSharpMiddlewareOptions> options)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public string GetExtensionFromUri(string uri)
         {
-            // TODO: Investigate using span to reduce allocations here.
-            string[] parts = uri.Split('?');
-            if (parts.Length > 1 && QueryHelpers.ParseQuery(parts[1]).TryGetValue(FormatWebProcessor.Format, out StringValues ext))
+#if !NETCOREAPP3_1_OR_GREATER
+            uri = uri.ToLowerInvariant();
+#endif
+
+            int query = uri.IndexOf('?');
+            ReadOnlySpan<char> path;
+
+            if (query > -1)
+            {
+                if (QueryHelpers.ParseQuery(uri.Substring(query)).TryGetValue(FormatWebProcessor.Format, out StringValues ext))
```

Comment:
Parsing can be skipped if the string doesn't contain `FormatWebProcessor.Format.ToLowerInvariant()`, might be worth it since otherwise it's creating a Dictionary on every call, and a new string with `Substring()`.